### PR TITLE
Enable layout debugging for `impl Trait` type aliases

### PR DESCRIPTION
**Pull request by RalfJung**
I also made it print the actual type name that the alias picks under the hood.

### CODE REVIEW

File: src/librustc_passes/layout_test.rs [hunk 1]
```diff
@@ -84,7 +85,7 @@ impl LayoutTest<'tcx> {
                         sym::debug => {
                             self.tcx.sess.span_err(
                                 item.span,
-                                &format!("layout debugging: {:#?}", *ty_layout),
+                                &format!("layout debugging for type {:?}: {:#?}", ty, *ty_layout),
```
Comment (eddyb) [inline src/librustc_passes/layout_test.rs:88]: Wondering if maybe this should be "layout for type" without "debugging".
Or `"layout_of({}) = {:#?}"` :P.

Comment (eddyb) [reply src/librustc_passes/layout_test.rs:88]: (but this PR is rolled up so this is for another time)

**Comment by eddyb** [comment]
@RalfJung is there an issue for moving some/all `print-type-sizes` tests to the new style?

**Comment by RalfJung** [comment]
@eddyb yes: https://github.com/rust-lang/rust/issues/69852

File: src/librustc_passes/layout_test.rs (hunk 1 above)
Comment (Centril) [reply src/librustc_passes/layout_test.rs:88]: Would be good to have some backticks if not that.

**Pull request merged**